pkg/defaultcomponents/defaults_test.go

```diff
@@ -18,42 +18,79 @@ package defaultcomponents
 import (
 	"testing"
 
-	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
+const (
+	exportersCount  = 13
+	receiversCount  = 8
+	extensionsCount = 6
+	processorCount  = 12
+)
+
 func TestComponents(t *testing.T) {
 	factories, err := Components()
 	require.NoError(t, err)
 	exporters := factories.Exporters
+	require.Equal(t, exportersCount, len(exporters))
 	// aws exporters
-	assert.True(t, exporters["awsxray"] != nil)
-	assert.True(t, exporters["awsprometheusremotewrite"] != nil)
+	require.NotNil(t, exporters["awsxray"])
+	require.NotNil(t, exporters["awsemf"])
+	require.NotNil(t, exporters["awsprometheusremotewrite"])
 	// core exporters
-	assert.True(t, exporters["logging"] != nil)
-	assert.True(t, exporters["otlp"] != nil)
-	assert.True(t, exporters["otlphttp"] != nil)
+	require.NotNil(t, exporters["logging"])
+	require.NotNil(t, exporters["otlp"])
+	require.NotNil(t, exporters["otlphttp"])
 	// other exporters
-	assert.True(t, exporters["datadog"] != nil)
-	assert.True(t, exporters["dynatrace"] != nil)
-
-	assert.True(t, exporters["sapm"] != nil)
-	assert.True(t, exporters["signalfx"] != nil)
-	//assert.True(t, exporters["splunk_hec"] != nil)
```

Comment:
Is there a reason for the removed commented export or why this exporter is commented out?

Reply:
It was commented out because these exporters have not been cleared by the AWS security team, so they were removed. You can see that mentioned in
https://github.com/aws-observability/aws-otel-collector/blob/96a64ee8fbda3461efbf4173b0ea42a98264c4ec/pkg/defaultcomponents/defaults.go#L127-L130
I don't see a reason to have it in the tests until it's going to be included.